src/rust/bitbox02-rust/src/hal.rs

```diff
@@ -33,10 +34,15 @@ pub trait Sd {
     async fn write_bin(&mut self, filename: &str, dir: &str, data: &[u8]) -> Result<(), ()>;
 }
 
+pub trait Random {
+    fn random_32_bytes(&mut self) -> Box<zeroize::Zeroizing<[u8; 32]>>;
```

Comment:
Why boxing it up?

Reply:
because otherwise [u8;32] leaves copies on the stack everytime it moves. also it is just adopting the sig of the existig bitbox02::random::random_32_bytes function.